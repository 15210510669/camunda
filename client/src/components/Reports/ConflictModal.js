/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React, {useState} from 'react';
import {Modal, Button} from 'components';
import {t} from 'translation';

export default function ConflictModal({conflict, onClose, onConfirm}) {
  const [loading, setLoading] = useState(false);

  const confirm = async () => {
    setLoading(true);
    await onConfirm();
    setLoading(false);
  };

  return (
    <Modal open={conflict} onClose={onClose} onConfirm={confirm} className="ConflictModal">
      <Modal.Header>{t('report.saveConflict.header')}</Modal.Header>
      <Modal.Content>
        {conflict &&
          ['combined_report', 'alert'].map((type) => {
            if (conflict[type].length === 0) {
              return null;
            }

            return (
              <div key={type}>
                <p>{t(`report.saveConflict.${type}.header`)}</p>
                <ul>
                  {conflict[type].map(({id, name}) => (
                    <li key={id}>'{name || id}'</li>
                  ))}
                </ul>
                <p>
                  <b>{t(`report.saveConflict.${type}.message`)}</b>
                </p>
              </div>
            );
          })}
      </Modal.Content>
      <Modal.Actions>
        <Button main disabled={loading} className="close" onClick={onClose}>
          {t('saveGuard.no')}
        </Button>
        <Button main disabled={loading} primary className="confirm" onClick={confirm}>
          {t('saveGuard.yes')}
        </Button>
      </Modal.Actions>
    </Modal>
  );
}
