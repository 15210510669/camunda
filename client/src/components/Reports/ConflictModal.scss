.ConflictModal {
  ul {
    list-style: none;
    padding: 0;
  }

  .conflictMessage {
    margin: 0;
  }
}
