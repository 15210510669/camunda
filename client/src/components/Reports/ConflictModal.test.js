/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {shallow} from 'enzyme';
import ConflictModal from './ConflictModal';

const props = {
  conflict: {alert: [{id: '1', name: 'alert', type: 'alert'}], combined_report: []},
  onClose: jest.fn(),
  onConfirm: jest.fn(),
};

it('should match snapshot', () => {
  const node = shallow(<ConflictModal {...props} />);

  expect(node).toMatchSnapshot();
});
