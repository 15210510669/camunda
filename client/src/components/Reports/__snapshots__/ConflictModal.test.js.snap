// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Modal
  className="ConflictModal"
  onClose={[MockFunction]}
  onConfirm={[Function]}
  open={
    Object {
      "alert": Array [
        Object {
          "id": "1",
          "name": "alert",
          "type": "alert",
        },
      ],
      "combined_report": Array [],
    }
  }
>
  <Component>
    Destructive Changes
  </Component>
  <Component>
    <div
      key="alert"
    >
      <p>
        These changes are incompatible with the Alert(s):
      </p>
      <ul>
        <li
          key="1"
        >
          '
          alert
          '
        </li>
      </ul>
      <p>
        <b>
          Warning: Saving will delete the Alert(s).
        </b>
      </p>
    </div>
  </Component>
  <Component>
    <ForwardRef(Button)
      className="close"
      disabled={false}
      main={true}
      onClick={[MockFunction]}
    >
      Don't Save
    </ForwardRef(Button)>
    <ForwardRef(Button)
      className="confirm"
      disabled={false}
      main={true}
      onClick={[Function]}
      primary={true}
    >
      Save Changes
    </ForwardRef(Button)>
  </Component>
</Modal>
`;
